src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -970,7 +995,7 @@ private static string JoinCore(ReadOnlySpan<char> separator, ReadOnlySpan<string
                     }
```

Comment:
`ThrowOutOfMemoryException` can switched to use the new helper with better message.